### PR TITLE
Alp 9 tab component

### PR DESCRIPTION
**Pull request by Leonlui**

### CODE REVIEW

File: aunt-leahs-app/src/components/tab/tab.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]: Use destructuring to import Component.
```
import React, { Component } from 'react';
```

File: aunt-leahs-app/src/components/tab/tab.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import PropTypes from 'prop-types';
+import AppBar from '@material-ui/core/AppBar';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import Typography from '@material-ui/core/Typography';
+import Box from '@material-ui/core/Box';
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]: Use destructuring to import components from material-ui.
```
import { AppBar, Tabs, Tab, Typography, Box } from '@material-ui/core';
```

File: aunt-leahs-app/src/components/tab/tab.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+{/* <TabComponent
+        name = "Tab"
+        items={[
+            { title: 'First Tab', content: <h1>hello</h1>},
+            { title: 'Second Tab',content: <CustomButton size="small" color="primary">I am a button</CustomButton>},
+            { title: 'Third Tab', content: 'Some other class'}
+            ...
+        ]} /> */}
+
+function TabPanel(props) {
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]: Can you explain the purpose of this function and why it's defined outside of the class?

File: aunt-leahs-app/src/components/tab/tab.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    return (
+        <Typography
+            component="div"
+            role="tabpanel"
+            hidden={value !== index}
+            id={`full-width-tabpanel-${index}`}
+            aria-labelledby={`full-width-tab-${index}`}
+            {...other}
+        >
+            <Box p={3}>{children}</Box>
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]: This is a magic number code smell. What is 3? This should probably be replaced by a variable.

File: aunt-leahs-app/src/components/tab/tab.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            value: 0,
+            items: this.props.items || [],
+        }
+    }
+
+    handleChange = (event, newValue) => {
+        this.setState({ value: newValue })
+    };
+
+    render () {return <div>
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]: return and div should both be on a new line.
```
render() {
     return (
          <div>
          ...
          </div>
     )
}

File: aunt-leahs-app/src/components/tab/tab.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        <AppBar position="static" color="default">
+            <Tabs
+                value={this.state.value}
+                onChange={this.handleChange}
+                indicatorColor="primary"
+                textColor="primary"
+                variant="fullWidth"
+                aria-label="tab label"
+            >
+            {this.state.items.map(item =>
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]: Use a more descriptive naming convention than items and item. What are items?

File: aunt-leahs-app/src/components/tab/tab.js (hunk 4 above)
Comment (Leonlui) [reply aunt-leahs-app/src/components/tab/tab.js]: It is a padding for the tab content, i have created a variable name for it

File: aunt-leahs-app/src/components/tab/tab.js (hunk 3 above)
Comment (Leonlui) [reply aunt-leahs-app/src/components/tab/tab.js]: Dissolved the classed and added it to the tab component class

File: aunt-leahs-app/src/components/tab/tab.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+// USAGE:
+{/* <TabComponent
+        tabs={[
+            { title: 'First Tab', content: <h1>hello</h1>},
+            { title: 'Second Tab',content: <CustomButton size="small" color="primary">I am a button</CustomButton>},
+            { title: 'Third Tab', content: 'Some other class'}
+            ...
+        ]} /> */}
+
+class TabComponent extends React.Component{
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]:
```suggestion
class TabComponent extends Component {
```

File: aunt-leahs-app/src/components/tab/tab.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    constructor(props){
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]:
```suggestion
    constructor(props) {
```

File: aunt-leahs-app/src/components/tab/tab.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        super(props);
+        this.state = {
+            value: 0,
+            tabs: this.props.tabs || [],
+        }
+    }
+    topLeftPadding =2;
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]: This seems out of place. If this is a variable, then you can declare it in the render function before the return. If it's styling, then you should import it from the styles.css file.

File: aunt-leahs-app/src/components/tab/tab.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+    handleChange = (event, newValue) => {
+        this.setState({ value: newValue })
+    };
+
+    render () {
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab/tab.js]:
```suggestion
    render() {
```

**Review by codygagnon — CHANGES_REQUESTED** [review]
This looks good, Leon. There are just a few small changes needed before this PR can be merged.

**Comment by thaingocnguyen** [comment]
Tab.js should be renamed to tab.jsx should be moved to the components instead of being in a subfolder just so it's consistent.

File: aunt-leahs-app/src/components/tab.jsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+            tabs: this.props.tabs || [],
+        }
+    }
+
+    handleChange = (event, newValue) => {
+        this.setState({ value: newValue })
+    };
+
+    render() {
+        var topLeftPadding =2;
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab.jsx]:
```suggestion
        var topLeftPadding = 2;
```

File: aunt-leahs-app/src/components/tab.jsx [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+            tabs: this.props.tabs || [],
+        }
+    }
+
+    handleChange = (event, newValue) => {
+        this.setState({ value: newValue })
+    };
+
+    render() {
+        var topLeftPadding = 2;
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab.jsx]:
```suggestion
        const topLeftPadding = 2;
```

File: aunt-leahs-app/src/components/tab.jsx [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+                        value={this.state.value}
+                        onChange={this.handleChange}
+                        indicatorColor="primary"
+                        textColor="primary"
+                        variant="fullWidth"
+                        aria-label="tab label"
+                    >
+                    {this.state.tabs.map(tab =>
+                        <Tab label={tab.title} />
+                    )}
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab.jsx]: Fix indentation.

File: aunt-leahs-app/src/components/tab.jsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+                {this.state.tabs.map((tab, index) =>
+                        <Typography
+                            component="div"
+                            role="tabpanel"
+                            hidden={this.state.value !== index}
+                            id={`full-width-tabpanel-${index}`}
+                            aria-labelledby={`full-width-tab-${index}`}
+                        >
+                            <Box p={topLeftPadding}>{tab.content}</Box>
+                        </Typography>
```
Comment (codygagnon) [inline aunt-leahs-app/src/components/tab.jsx]: Fix indentation (<Typography should line up with </Tabs> in line 45).

**Review by codygagnon — APPROVED** [review]

**Review by michaelfromyeg — APPROVED** [review]
LGTM

**Pull request merged**